target/linux/ath79/image/generic-compex.mk

```diff
@@ -0,0 +1,36 @@
+include ./common-compex.mk
+
+define Device/compex_wpj531
+  $(Device/compex_wpj-16m)
+  DEVICE_TITLE := Compex WPJ531 (16MB flash)
+  COMPEX_VENDOR_ID := 0x11f6
+  BOARDNAME := WPJ531
+endef
+TARGET_DEVICES += compex_wpj531
+
+define Device/compex_wpj531-7A03
```

Comment:
is this the preferred way of creating different factory images?

Reply:
What do you mean by "creating different factory images"?

BTW: I'd recommend to put the compex devices into generic.mk with all the others, unless you are expecting to have support for > 10 of them.

Reply:
e.g. the WPJ531 has at least 4 different board revisions (7A03, 7A04, 7A06, 7A07).
all revisions support the same sysupgrade file.
but the cpximg formats are slightly different..

so, to support the different revisions we would need:
```
wpj531-7A03-cpximg.bin
wpj531-7A04-cpximg.bin
wpj531-7A06-cpximg.bin
wpj531-7A07-cpximg.bin
```
the sysupgrade files would be identical for all of them.

i'll move it to generic.mk.

Reply:
Maybe have a look at the TP-Link TL-WR841 v11/12.
The had different factory images by region:
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/image/tiny-tp-link.mk#L177

Maybe you can work out a similar solution where you just specify the revision by a parameter to your build code.

Reply:
good point.
i can envision that working.
then i should probably add the boards to mkmylofw.

this might take a while longer.

Reply:
Maybe also scan through generic.mk whether you find other similar solutions, might make your job faster ...

Reply:
new commit
i'll resolve this conversation.